### ISSUE TITLE
False positive in rule `vertical_parameter_alignment_on_call`

### ISSUE THREAD

**Issue by NikKovIos** [realm/SwiftLint#6467]
version 0.63.0

```
vertical_parameter_alignment_on_call:
    severity: error
```

example to reproduce:
```
    func test() {
        СreateAdUpdateShopHelper.updateShopModels(categoryModels: &categoryModels,
                                                  contactModels: &contactModels, // < - error
                                                  country: response.country!, // < - error
                                                  contact: response.contact, // < - error
                                                  shop: response.shop) // < - error
    }
```

**Comment by SimplyDanny** [comment]
You intentionally used a `U+0421` "C" as the first character?

Even if it was a mistake, you are highlighting an important shortcoming of the rule: it assumes every character to have length 1 ...

**Comment by NikKovIos** [comment]
It was a mistake, thanks

**Comment by theamodhshetty** [comment]
I picked this up in #6535.

The fix converts the alignment comparison to character columns before checking subsequent parameters, so a multi-byte character earlier on the first line no longer shifts the reference column and produces false positives. I also added a regression example based on the reported case.
